xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/flamingo.js

```diff
@@ -40,14 +40,87 @@ require(['jquery', 'bootstrap'], function($) {
         }
       }
     });
-
-    // When the drawer is close, collapse sub items
-    $(body).on('drawer.closed', function() {
-      $('.drawer-menu-sub-item').removeClass('in').attr('aria-expanded', 'false');
-    });
-
     // Activate the popover when hovering the Translate button.
     var translateButton = $('#tmTranslate [data-toggle="popover"]');
     translateButton.attr('title', translateButton.attr('data-title')).popover();
   });
 });
+
+/*
+  Handle the behavior of all drawers on the page.
+  In order to create a drawer that is compatible with this script, two elements are necessary:
+  * A drawer toggler button, that will have:
+   ** a reference to the drawer container in its attribute 'aria-control'
+   ** the class 'drawer-toggle'
+  * A drawer container, that will have:
+   ** a unique ID
+   ** the class 'drawer-nav' (mostly for style) 
+   ** the class 'closed' if the drawer is supposed to start in a closed state.
+   ** a label 
+   ** The content to display inside. It's asserted that at least one element in this content can receive focus.
+  For an example of drawer creation, see #tmDrawerActivator and #tmDrawer.
+ */
+require(['jquery'], function($) {
+  // The overlay is the same whatever the drawer opened.
+  const drawerOverlay = $(".drawer-overlay");
+  $('.drawer-toggle').each(function(index) {
+    // Setting up the drawer.
+    let drawerContainerToggler = $(this);
+    let drawerId = drawerContainerToggler.attr('aria-controls');
+    let drawerContainer = $('#'+drawerId);
```

Comment:
As this is supposed a re-usable component, I think this should properly escape the id for the use in a selector. Alternatively, it could also use the native plain JavaScript method to get an element by id which then wouldn't require escaping. Also missing, spaces around `+`.

Reply:
:+1: Addressed in https://github.com/xwiki/xwiki-platform/pull/2282/commits/1e58216ef0c9138855bfb832ef2a9f5ee482a767